plugins/volumeFilter_test.go

```diff
@@ -1126,8 +1157,19 @@ func runTestVolumeFilterFn(
 
 func makeSellOpAmtPrice(amount float64, price float64) *txnbuild.ManageSellOffer {
 	return &txnbuild.ManageSellOffer{
-		Buying:  txnbuild.NativeAsset{},
-		Selling: txnbuild.NativeAsset{},
+		Buying:  testQuoteAsset,
+		Selling: testBaseAsset,
+		Amount:  fmt.Sprintf("%.7f", amount),
+		Price:   fmt.Sprintf("%.7f", price),
+	}
+}
+
+func makeBuyOpAmtPrice(amount float64, price float64) *txnbuild.ManageSellOffer {
```

Comment:
I'm a little confused about this function, rest looks perfect (buy offers are tricky!)

here we are selling `amount` units of of the selling asset, i.e. of the quote asset. But what we want is for the `amount` to represent the _units of the base asset_. To achieve this we will need to do some math in this function to figure out what value to set as the amount field on mso (which represents quote units) so that the base units is equal to the `amount` input.

I ran this using the buysell strategy with the amount set to 1000.0 units (of base on either side), with the buy op being displayed first:

```
2020/12/23 02:35:59 submitting the following ops
2020/12/23 02:35:59 &{Selling:{Code:COUPON Issuer:GBMMZMK2DC4FFP4CAI6KCVNCQ7WLO5A7DQU7EC7WGHRDQBZB763X4OQI} Buying:{} Amount:163.4735274 Price:6.1171984 OfferID:0 SourceAccount:0xc00047e5e0}
2020/12/23 02:35:59 &{Selling:{} Buying:{Code:COUPON Issuer:GBMMZMK2DC4FFP4CAI6KCVNCQ7WLO5A7DQU7EC7WGHRDQBZB763X4OQI} Amount:1000.0000000 Price:0.1638006 OfferID:0 SourceAccount:0xc00047e6a0}
```

above, we can see that the selling operation (second op) has amount = 1000.0 and price = 0.16 which is correct
the buying op (first op) has amount set to 163.xx and price set to 6.xx
we need to replicate this.

To do so, we need to:
- set our Amount field on the manageSellOffer here equal to `amount * price` (so we get 1000 * 0.16 ~= 163.xx like displayed above)
- set the Price field to `1/price` (so we get 1/0.16 ~= 6.xx like displayed above)

Going by the [definition of ManageSellOffer in the docs][definition-mso] price represents 1 unit of selling in terms of buying, or 1 unit of quote in terms of base here.
In our test, at a price of XLM/QUOTE of 0.16xx we want to buy 1000 units of base (XLM) and sell 163.xx units of quote (QUOTE).
This translates to an MSO with:
- amount = 163.xx (units of selling asset being sold)
- price = 1000/163 => 6.xx (1 unit of selling in terms of buying)
This matches what we got in our calculations above.

So to summarize, we should set the `Amount` field on MSO equal to `amount * price` and Price field to `1/price`

does this make sense?

[definition-mso]: https://www.stellar.org/developers/guides/concepts/list-of-operations.html#manage-sell-offer

Reply:
PS: please can you add the commented code sample above until the end of the first set of bullet points into the code as a comment. This will make sure we save this thinking for future reference next to the code itself. thanks!

Reply:
Nice, this makes perfect sense! As discussed, this breaks a bunch of buy-side tests. I think this is because the test cases are now in a weird boundary condition, where they meet the specified condition for the sell side but not the buy side. I'm not sure if this requires tweaking the test case or making  `wantBuy`/`wantSell`  values for Amount and Price.